cinema/views.py

```diff
@@ -1,11 +1,14 @@
 from datetime import datetime
 
 from django.db.models import F, Count
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticated
```

Comment:
The import statement for `IsAuthenticated` is correct, but ensure that the custom permissions from `permissions.py` are also imported and applied to the relevant viewsets as per the task requirements.